src/ATen/native/xpu/sycl/HistogramddKernels.cpp

```diff
@@ -186,24 +238,74 @@ void histogramdd_kernel(
             ? weight.value().packed_accessor64<const scalar_t, 1>()
             : PackedTensorAccessor64<const scalar_t, 1>(
                   nullptr, self.sizes().data(), self.strides().data());
+
+        Tensor hist_strides_xpu = at::tensor(
+            hist.strides(),
+            self.options()
+                .dtype(c10::kLong)
+                .memory_format(at::MemoryFormat::Contiguous));
+        Tensor bin_edges_contig_ptr = at::empty(
+            {N},
+            self.options()
+                .dtype(c10::kUInt64)
+                .device(at::kCPU)
+                .memory_format(at::MemoryFormat::Contiguous));
```

Comment:
Can we follow PyTorch to make bin_edges contiguous, then we don't need create tensor accessor
```
    std::vector<Tensor> bin_edges_contig(bin_edges.size());
    for (const auto dim : c10::irange(bin_edges_contig.size())) {
        bin_edges_contig[dim] = bin_edges[dim].contiguous();
    }
```